diff --git a/include/boost/ut.hpp b/include/boost/ut.hpp
index 4810d9c..c0c9ed3 100644
--- a/include/boost/ut.hpp
+++ b/include/boost/ut.hpp
@@ -34,6 +34,7 @@ export import std;
 #if defined(__cpp_lib_format) or \
     (defined(_LIBCPP_VERSION) and _LIBCPP_VERSION >= 170000)
 #define BOOST_UT_HAS_FORMAT
+#include <fmt/core.h>
 #endif
 
 #if not defined(__cpp_rvalue_references)
@@ -2349,15 +2350,15 @@ struct log {
 #if defined(BOOST_UT_HAS_FORMAT)
 #if __cpp_lib_format >= 202207L
   template <class... Args>
-  void operator()(std::format_string<Args...> fmt, Args&&... args) {
+  void operator()(fmt::format_string<Args...> fmt, Args&&... args) {
     on<std::string>(
-        events::log{std::vformat(fmt.get(), std::make_format_args(args...))});
+        events::log{fmt::vformat(fmt.get(), fmt::make_format_args(args...))});
   }
 #else
   template <class... Args>
   void operator()(std::string_view fmt, Args&&... args) {
     on<std::string>(
-        events::log{std::vformat(fmt, std::make_format_args(args...))});
+        events::log{fmt::vformat(fmt, fmt::make_format_args(args...))});
   }
 #endif
 #endif
